### ISSUE TITLE
Error importing `JSONParser`

### ISSUE THREAD

**Issue by tegefaulkes** [juanjoDiaz/streamparser-json#25]
In our code base when we try to import `JSONParser` with `import { JSONParser } from '@streamparser/json';` we get the following error `TS2307: Cannot find module '@streamparser/json' or its corresponding type declarations.`.

Currently as a work around we are importing it with `const jsonStreamParsers = require('@streamparser/json');` which works fine.

My question is, do you have any insights to why the usual import is failing? Or is this likely a problem with the configuration of our project in some way?

Thanks.

* Related https://github.com/MatrixAI/Polykey/issues/516

**Comment by juanjoDiaz** [comment]
Hi @tegefaulkes ,

Which version of @streamparser/json are you using?
Which bundler are you using (webpack, rollup, esbuild, ...)?

The types are embedded in the package: https://github.com/juanjoDiaz/streamparser-json/blob/main/packages/plainjs/package.json#L8-L15 so there shouldn't be any problem.

You can check in your `node_modules` folder that the package is installed and the types are where they should be.

You can see [here](https://github.com/search?q=%40streamparser%2Fjson&type=code) that a bunch of people is using it without a problem.

**Comment by CMCDragonkai** [comment]
Our tsconfig specifies `"module": "CommonJS"`.

If we use a simple script like this:

```ts
import { JSONParser } from '@streamparser/json';

async function main() {
  console.log(JSONParser);
}

main();
```

Then attempt to compile with `tsc` it gives us this:

```
[nix-shell:~/Projects/Polykey]$ tsc ./test.ts
test.ts:1:28 - error TS2307: Cannot find module '@streamparser/json' or its corresponding type declarations.

1 import { JSONParser } from '@streamparser/json';
                             ~~~~~~~~~~~~~~~~~~~~


Found 1 error in test.ts:1
```

Current version of `@streamparser/json` is `0.0.12`.

The `tsc` is `4.9.3`.

We use lots of other packages, this is the first we have encountered this problem. Is it because of different package configuration?

**Comment by CMCDragonkai** [comment]
We don't use any bundler. It's just `tsc`.

We're also using `swc`, but `swc` does not have the same problem.

**Comment by CMCDragonkai** [comment]
Full `tsconfig.json` is:

```
{
  "compilerOptions": {
    "outDir": "./dist",
    "tsBuildInfoFile": "./dist/tsbuildinfo",
    "incremental": true,
    "sourceMap": true,
    "declaration": true,
    "allowJs": true,
    "strictNullChecks": true,
    "noImplicitAny": false,
    "experimentalDecorators": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "moduleResolution": "node",
    "module": "CommonJS",
    "target": "ES2022",
    "baseUrl": "./src",
    "paths": {
      "@": ["index"],
      "@/*": ["*"]
    },
    "noEmit": true
  },
  "include": [
    "./src/**/*",
    "./src/**/*.json",
    "./tests/**/*",
    "./scripts/**/*",
    "./benches/**/*"
  ],
  "ts-node": {
    "require": ["tsconfig-paths/register"],
    "transpileOnly": true,
    "swc": true
  }
}
```

**Comment by tk-o** [comment]
I have it fixed on my end.

As a hint, I went to the [`dist/package.json`](https://unpkg.com/browse/@streamparser/json-whatwg@0.0.12/package.json) file and removed this part:
```json
  "typesVersions": {
    "*": {
      "*": ["./src/*"]
    }
  },
```

The typings were immediately matched in my VS Code.

Not sure why the `typesVersions` prop is an issue, but apparently that's the case.

**Comment by CMCDragonkai** [comment]
Cool, are you going to release a new version with the fix?

**Comment by juanjoDiaz** [comment]
Thanks for the hint.

The issue seems caused by a bug in Typescript: https://github.com/microsoft/TypeScript/issues/43133

I'll do a fix asap 🙂

**Comment by juanjoDiaz** [comment]
This is fixed as part of 0.0.13.

Thanks for your patience!

**Comment by CMCDragonkai** [comment]
Nice!
